R/execution.r

```diff
@@ -180,12 +198,11 @@ execute = function(request) {
             # Only send a response when there is regular console output
             if (nchar(data[['text/plain']]) > 0) {
                 if (getOption('jupyter.rich_display')) {
-                    tryCatch({
-                        for (mime in getOption('jupyter.display_mimetypes')) {
-                            r <- mime2repr[[mime]](obj)
-                            if (!is.null(r)) data[[mime]] <- r
-                        }
-                    }, error = handle_error)
+                    for (mime in getOption('jupyter.display_mimetypes')) {
+                        tryCatch(r <- mime2repr[[mime]](obj),
+                            error = handle_display_error)
+                        if (!is.null(r)) data[[mime]] <- r
```

Comment:
If the repr fails, I think `r` will either be undefined, or have the value from the previous iteration of the loop. So I think this needs to be inside the tryCatch, so it only runs if the repr succeeds.